components/fonts/platform/freetype/font.rs

```diff
@@ -582,3 +537,23 @@ unsafe extern "C" {
         length: *mut FT_ULong,
     ) -> FT_Error;
 }
+
+enum FreeTypeFaceTableProviderData {
+    Web(FontData),
+    Local(Mmap, u32),
+}
+
+impl FreeTypeFaceTableProviderData {
+    fn font_ref(&self) -> Result<FontRef<'_>, ReadError> {
+        match self {
+            Self::Web(ipc_shared_memory) => FontRef::new(&ipc_shared_memory.0),
+            Self::Local(mmap, index) => FontRef::from_index(mmap, *index),
+        }
+    }
+}
+
+impl std::fmt::Debug for FreeTypeFaceTableProviderData {
+    fn fmt(&self, _: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Ok(())
```

Comment:
Is this intentionally a no-op debug implementation? Do we not want to distinguish at least between `Web` and `Local` versions of `self`?